### PR TITLE
Plot analysis for visualising evaluation metrics

### PR DESCRIPTION
**Pull request by Soniyanayak51**
PR for issue #6 

1. Simple scatter plot for x and y values
2. Violin and Box plots
A boxplot is a standardized way of displaying the dataset based on a five-number summary: the minimum, the maximum, the sample median, and the first and third quartiles. I have changed the median value of box plot to mean value(average of y).
Violin plot is similar but it also presents the probability density.

@dzeber please let me know if the approach is on the right track!

### CODE REVIEW

**Comment by Soniyanayak51** [comment]
@dzeber I would like to try dot/strip plots or python representation of R’s ggplot2 for this if you could review this once

File: dev/Soniyanayak51/issue6/evalMetric.ipynb [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    "from sklearn.utils import shuffle\n",
+    "from sklearn.metrics import confusion_matrix\n",
+    "import seaborn as sns\n",
+    "import matplotlib.gridspec as gridspec\n",
+    "from sklearn.preprocessing import StandardScaler\n",
+    "from sklearn.manifold import TSNE\n",
+    "from pandas import set_option\n",
+    "from sklearn.svm import SVC\n",
+    "from sklearn.linear_model import LogisticRegression\n",
+    "from sklearn.datasets import load_iris"
```
Comment (dzeber) [inline dev/Soniyanayak51/issue6/evalMetric.ipynb]: Please only import things you use

File: dev/Soniyanayak51/issue6/moduleForEvalMetric.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+from sklearn.utils import shuffle
+from sklearn.metrics import confusion_matrix
+import seaborn as sns
+import matplotlib.gridspec as gridspec
+from sklearn.preprocessing import StandardScaler
+from sklearn.manifold import TSNE
+from pandas import set_option
+from sklearn.svm import SVC
+from sklearn.linear_model import LogisticRegression
+from sklearn.datasets import load_iris
```
Comment (dzeber) [inline dev/Soniyanayak51/issue6/moduleForEvalMetric.py]: Please only import things you use

**Comment by Soniyanayak51** [comment]
@dzeber thanks for the great interesting suggestions. Will implement them

File: dev/Soniyanayak51/issue6/moduleForEvalMetric.py (hunk 2 above)
Comment (Soniyanayak51) [reply dev/Soniyanayak51/issue6/moduleForEvalMetric.py]: Will keep this in mind

**Comment by Soniyanayak51** [comment]
@dzeber @mlopatka I have generated a small dataset by changing C values and generating accuracy for each fold using repeated K-fold. I have added a new plot to visualize variance.
Plots are plotted with the above data.
Please review!

**Comment by mlopatka** [comment]
@Soniyanayak51 It appears that you have multiple PRs addressing some of the same changes to your work. Can you please follow up with comments in each of your current outstanding PRs as to which of them are active work and which of them are stale.

It would be very helpful to know where we should focus our attention to ensure that reviews are on the most recent work. Feel free to close out stale PRs that you submitted so that we don't review them.

**Comment by Soniyanayak51** [comment]
@mlopatka I have been using the same notebook as my startup task to generate the data here. I have used the same notebook and the module from PR #22 in most of my work as suggested by @dzeber. 
Do I need to remove it and make new modules?
Also could you please review the latest changes I made in this PR? I have generated a small dataset by changing C values and generating accuracy for each fold using repeated K-fold. Thank you:)

**Comment by dzeber** [comment]
@Soniyanayak51 using the same notebook and module as #22 is fine. When these eventually get merged, they should appear as incremental changes to the same files. 

Same comment as in #22, please use the CSV file included in the repo.

I like that you are trying out different approaches to the visualization, as they are more or less appropriate in different situations. One suggestion I would make for all of them is to plot the x-values at equally spaced intervals. Using the original numerical scale causes those uneven gaps which is difficult to read. The log-scale for the mean-variance plot helps but it is still difficult to read because the log is base 10 rather than 2. I think the scale of the differences between the parameter values on the x-axis is not so important to show in these plots, as the goal is to select whichever one has the optimal metric value.

As the goal of the task is to create a function that can be applied to a general CV dataset and metric, the next step would be to settle on a final visualization approach based on what you've tried and develop it into a generalized function. Note that the box/violin and mean/variance plots are good ways to summarize when you have many y values for each x, but can be misleading when the dataset is so small (in which case the simple plot may be more representative).
